bindings/rust/integration/tests/internet_http_client.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Purpose: ensure that s2n-tls is compatible with other http/TLS implementations
+///
+/// This test uses s2n-tls-hyper to make http requests over a TLS connection to
+/// a number of well known http sites.
+#[cfg(feature = "network-tests")]
+mod http_get {
+    use bytes::Bytes;
+    use http::{StatusCode, Uri};
+    use http_body_util::{BodyExt, Empty};
+    use hyper_util::{client::legacy::Client, rt::TokioExecutor};
+    use s2n_tls::{config::Config, security};
+    use s2n_tls_hyper::connector::HttpsConnector;
+    use std::{error::Error, str::FromStr};
+    use tracing_subscriber::filter::LevelFilter;
+
+    #[derive(Debug)]
+    struct TestCase {
+        pub query_target: &'static str,
+        pub expected_status_code: u16,
+    }
+
+    impl TestCase {
+        const fn new(domain: &'static str, expected_status_code: u16) -> Self {
+            TestCase {
+                query_target: domain,
+                expected_status_code,
+            }
+        }
+    }
+
+    const TEST_CASES: &[TestCase] = &[
+        // Akamai hangs indefinitely. This is also observed with curl and chrome
+        // https://github.com/aws/s2n-tls/issues/4883
+        // TestCase::new("https://www.akamai.com/", 200),
+
+        // this is a link to the s2n-tls unit test coverage report, hosted on cloudfront
+        TestCase::new("https://dx1inn44oyl7n.cloudfront.net/main/index.html", 200),
+        // this is a link to a non-existent S3 item
+        TestCase::new("https://notmybucket.s3.amazonaws.com/folder/afile.jpg", 403),
+        TestCase::new("https://www.amazon.com", 200),
+        TestCase::new("https://www.apple.com", 200),
+        TestCase::new("https://www.att.com", 200),
+        TestCase::new("https://www.cloudflare.com", 200),
+        TestCase::new("https://www.ebay.com", 200),
+        TestCase::new("https://www.google.com", 200),
+        TestCase::new("https://www.mozilla.org", 200),
+        TestCase::new("https://www.netflix.com", 200),
+        TestCase::new("https://www.openssl.org", 200),
+        TestCase::new("https://www.t-mobile.com", 200),
+        TestCase::new("https://www.verizon.com", 200),
+        TestCase::new("https://www.wikipedia.org", 200),
+        TestCase::new("https://www.yahoo.com", 200),
+        TestCase::new("https://www.youtube.com", 200),
+        TestCase::new("https://www.github.com", 301),
+        TestCase::new("https://www.samsung.com", 301),
+        TestCase::new("https://www.twitter.com", 301),
+        TestCase::new("https://www.facebook.com", 302),
+        TestCase::new("https://www.microsoft.com", 302),
+        TestCase::new("https://www.ibm.com", 303),
+        TestCase::new("https://www.f5.com", 403),
+    ];
+
+    #[tokio::test]
+    async fn http_get_test() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
I'm not seeing any retry mechanism-- is that handled by hyper automatically, or do you plan to add that in another PR? How flaky are these tests? Same for the PQ tests, since s2n-tls-tokio definitely doesn't handle retries.

Reply:
I'll add more information about this in the callouts, but 
1. it is not handled by hyper automatically
2. I do plan on adding it in another PR, but not until actually observing some failures.
3. I don't have any concrete data, but I have not seen any failures locally.

We don't currently have a good understanding of whether the flakiness comes from. TCP? TLS? DNS? Being able to actually observe these failures will allow us to
1. document where the flakiness actually comes from
2. have tightly scoped retries to catch other (real) transient errors.

Reply:
I don't expect that, but I certainly wouldn't be surprised by that outcome 🤷 .

I'm also wondering whether the amount of queries that we were doing (e.g. 30 failed TLS handshakes for each domain per test run) might have led some domains to start refusing connections.